js/models/site.es6.js

```diff
@@ -15,11 +15,26 @@ function Site (attrs) {
 Site.prototype = $.extend({},
   Parent.prototype,
   {
-      toggleWhitelist: function (s) {
-          console.log('Site toggleWhitelist() not implemented');
-          this.isWhitelisted = !this.isWhitelisted;
+      toggleWhitelist: function () {
+          let site = this._getSite();
+          if(site){
+              site.toggleWhiteList();
+              this.isWhitelisted = !this.isWhitelisted;
+          }
+      },
+
+      setWhitelistStatus: function() {
+          let site = this._getSite();
+          if(site){
+            this.isWhitelisted = site.isWhiteListed();
+          }
+      },
 
-          // TODO actually update whitelist
+      _getSite: function() {
+          let tab = backgroundPage.tabs[this.tabId];
+          let host = backgroundPage.utils.extractHostFromURL(tab.url);
+          let site = backgroundPage.Sites.get(host);
+          return site;
```

Comment:
do we have to do this every time ? Isn't there a current site object that already has this info?